### PR TITLE
Fix remaining dead links

### PR DESCRIPTION
**Pull request by julienrf**
See e.g. https://github.com/scala/scala-lang/runs/5353232371?check_suite_focus=true

### CODE REVIEW

File: blog/_posts/2017-10-16-fourth-dotty-milestone-release.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ You can learn more about Dotty on our [website](https://dotty.epfl.ch).
 
 <!--more-->
 
-This is our fourth scheduled release according to our [6-week release schedule](https://dotty.epfl.ch/docs/usage/version-numbers.html).
+This is our fourth scheduled release according to our [6-week release schedule](https://dotty.epfl.ch/docs/contributing/procedures/release.html).
```
Comment (julienrf) [inline blog/_posts/2017-10-16-fourth-dotty-milestone-release.md:21]: The page version-numbers says its content is outdated: https://github.com/lampepfl/dotty/blob/fa08d004308ff4dee799aafa30939e0647018d72/docs/_docs/usage/version-numbers.md

File: blog/_posts/2018-10-04-scaladoc-tables.md [hunk 2]
```diff
@@ -97,7 +97,7 @@ For a smorgasbord of examples take a look at [examples](https://janekdb.github.i
 
 ### Dotty and Scala 3
 
-With `dotty` now boosting its share of the development horizon it's important to place GFM into a [dottydoc](https://dotty.epfl.ch/docs/usage/dottydoc.html) and hence Scaladoc 3 context.
+With `dotty` now boosting its share of the development horizon it's important to place GFM into a [dottydoc](https://docs.scala-lang.org/scala3/guides/scaladoc/) and hence Scaladoc 3 context.
```
Comment (julienrf) [inline blog/_posts/2018-10-04-scaladoc-tables.md:100]: The page dottydoc says its content is outdated: https://github.com/lampepfl/dotty/blob/fa08d004308ff4dee799aafa30939e0647018d72/docs/_docs/usage/dottydoc.md

File: blog/_posts/2017-10-16-fourth-dotty-milestone-release.md (hunk 1 above)
Comment (bishabosha) [reply blog/_posts/2017-10-16-fourth-dotty-milestone-release.md:21]: this release procedure is also old, maybe we should just drop the link, as the release procedure from 2017 is not crucial to preserve

Comment (bishabosha) [inline blog/_posts/2017-10-16-fourth-dotty-milestone-release.md:21]:
```suggestion
This is our fourth scheduled release according to our 6-week release schedule.
```

Comment (julienrf) [reply blog/_posts/2017-10-16-fourth-dotty-milestone-release.md]: Done in 3af9bfe

**Pull request merged**